src/Session.php

```diff
@@ -2414,4 +2420,32 @@ public static function getCurrentProfile(): Profile
 
         return $profile;
     }
+
+    /**
+     * Runs a callable with the right checks disabled.
+     * @return mixed|void The return value of the callable.
+     * @throws Throwable Any throwable that was caught from the callable if any.
+     */
+    public static function callAsSystem(callable $fn)
+    {
+        $caught_throwable = null;
+        try {
+            self::$bypass_right_checks = true;
+            return $fn();
+        } catch (Throwable $e) {
+            $caught_throwable = $e;
+        } finally {
+            self::$bypass_right_checks = false;
+        }
+        throw $caught_throwable;
```

Comment:
`$caught_throwable` could be null here.
```suggestion
        throw $caught_throwable;
```

Reply:
My bad, there is a retyurn statement if there is no failure.